nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GenerateTableFetch.java

```diff
@@ -259,6 +261,11 @@ public void onTrigger(final ProcessContext context, final ProcessSessionFactory
                 }
             });
 
+            if(customWhereClause != null) {
+                // adding the custom WHERE clause (if defined) to the list of existing clauses.
+                maxValueClauses.add(customWhereClause);
```

Comment:
What do you think about forcing custom where clauses to be surrounded with parenthesis?
I worry that just putting it in like this may cause unexpected consequences if the where clauses includes OR's.